encode/vp9enc.c

```diff
@@ -658,9 +666,24 @@ vp9enc_create_encode_pipe(FILE *yuv_fp)
     vaQueryConfigEntrypoints(va_dpy, vp9enc_context.profile, entrypoints,
                              &num_entrypoints);
 
-    for	(slice_entrypoint = 0; slice_entrypoint < num_entrypoints; slice_entrypoint++)
-        if (entrypoints[slice_entrypoint] == VAEntrypointEncSlice)
-            break;
+    for (slice_entrypoint = 0; slice_entrypoint < num_entrypoints; slice_entrypoint++) {
```

Comment:
is it possible when both entrypoint[0] and entrypoint[1] are not  VAEntrypointEncSlice or VAEntrypointEncSliceLP? That will lead select_entrypoint always -1.

Reply:
entrypoints[] contains all supported entrypoints returned from the underlying driver, so it is possible both entrypoint[0] and entrypoint[1] are not VAEntrypointEncSlice or VAEntrypointEncSliceLP even on KBL. 

I guess you mean the all elements in entrypoints[]. Yes select_entrypoint will be -1 if the underlying driver doesn't  support VP9 encoding and user doesn't specify the encoding mode.  slice_entrypoint will be equal to num_entrypoints after the 'for' block and the application will exit by assert(0), so select_entrypoint(-1) won't cause any new issue.

Reply:
that make sense to me.